src/Interpreters/DDLWorker.cpp

```diff
@@ -494,14 +495,30 @@ void DDLWorker::parseQueryAndResolveHost(DDLTask & task)
             {
                 if (found_exact_match)
                 {
-                    throw Exception("There are two exactly the same ClickHouse instances " + address.readableString()
-                        + " in cluster " + task.cluster_name, ErrorCodes::INCONSISTENT_CLUSTER_DEFINITION);
+                    if(default_database == address.default_database)
+                    {
+                        throw Exception(
+                            "There are two exactly the same Snowball instances " + address.readableString() + " in cluster "
+                                + task.cluster_name, ErrorCodes::INCONSISTENT_CLUSTER_DEFINITION);
+                }
+                    else  ///circular replication is used.
```

Comment:
What does mean `circular replication`?

Reply:
https://www.altinity.com/blog/2018/5/10/circular-replication-cluster-topology-in-clickhouse